### ISSUE TITLE
Rule proposal: no-unused-style

### ISSUE THREAD

**Issue by golopot** [vuejs/eslint-plugin-vue#888]
**Please describe what the rule should do:**
Warn on classes/id/tag defined in scoped css but no used in template.

**What category should the rule belong to?**
[x] Warns about a potential error

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
  <input class="foo"/>
</template>

<style scoped>
  .not-foo { color: red;}
/* ~~~~~~~ vue/no-unused-style: `not-foo` is unused. */
</style>
```
**Additional context**

Problems:

1. Support for styles loaded from another file like `<style scoped src="./foo.css">`.

2. Support for css modules.

3. Support for sass/less.

4. Maybe it make sense offer this rule in another eslint plugin.

**Comment by ota-meshi** [comment]
Hi. 
I created this rule with another ESLint plugin.
https://future-architect.github.io/eslint-plugin-vue-scoped-css/rules/no-unused-selector.html
